src/lib/quoteEngine.js

```diff
@@ -185,6 +191,37 @@ const calculateOptimalPoolAllocation = (coverAmount, pools, useFixedPrice = fals
   return allocations;
 };
 
+/**
+ * Allocates a given amount to pools based on a custom pool ID priority list.
+ *
+ * @param {number} amountToAllocate - The amount to be allocated.
+ * @param {Array<object>} poolsData - An array of pool data objects.
+ * @param {Array<number>} customPoolIdPriority - A list of pool IDs in priority order.
+ * @return {object} An object containing the allocations by pool ID. (poolId => BigNumber amount)
+ */
+const customAllocationPriorityFixedPrice = (amountToAllocate, poolsData, customPoolIdPriority) => {
+  const allocations = {};
+  let coverAmountLeft = amountToAllocate;
+
+  while (coverAmountLeft > 0 && customPoolIdPriority.length > 0) {
+    const poolId = customPoolIdPriority.shift();
+    const pool = poolsData.find(poolData => poolData.poolId === poolId);
+
+    const capacityLeft = pool.totalCapacity.sub(pool.initialCapacityUsed);
+    const poolAllocation = bnMin(capacityLeft, coverAmountLeft);
+
+    allocations[poolId] = allocations[poolId] ? allocations[poolId].add(poolAllocation) : poolAllocation;
```

Comment:
nit: you don't need this check, since it will never go through the same pool more than once